doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/docUtils.kt

```diff
@@ -0,0 +1,446 @@
+        .removeSuffix("}")
+        .removePrefix("@link")
+        .trim()
+        .replace('#', '.')
+        .replace(Regex("""\(.*\)"""), "")
+        .trim()
+
+/**
+ * Get tag name from the start of some content.
+ * Can handle both
+ * `  @someTag someContent`
+ * and
+ * `{@someTag someContent}`
+ * and will return "someTag" in these cases.
+ */
+fun DocContent.getTagNameOrNull(): String? =
+    takeIf { it.trimStart().startsWith('@') || it.startsWith("{@") }
+        ?.trimStart()
+        ?.removePrefix("{")
+        ?.removePrefix("@")
+        ?.takeWhile { !it.isWhitespace() }
+
+/**
+ * Split doc content in blocks of content and text belonging to tags.
+ * The tag, if present, can be found with optional leading spaces in the first line of the block.
+ * You can get the name with [String.getTagNameOrNull].
+ * Splitting takes `{}`, `[]`, `()`, and triple backticks into account.
+ * Block "marks" are ignored if "\" escaped.
+ * Can be joint with '\n' to get the original content.
+ */
+fun DocContent.splitDocContentPerBlock(): List<DocContent> {
+    val docContent = this@splitDocContentPerBlock.split('\n')
+    return buildList {
+        var currentBlock = ""
+        val blocksIndicators = mutableListOf<Char>()
+        for (line in docContent) {
+
+            // start a new block if the line starts with a tag and we're not
+            // in a {} or `````` block
+            if (line.trimStart().startsWith("@") && blocksIndicators.isEmpty()) {
+                if (currentBlock.isNotEmpty()) add(currentBlock.removeSuffix("\n"))
+                currentBlock = "$line\n"
+            } else if (line.isEmpty() && blocksIndicators.isEmpty()) {
+                currentBlock += "\n"
+            } else {
+                if (currentBlock.isEmpty()) {
+                    currentBlock = "$line\n"
+                } else {
+                    currentBlock += "$line\n"
+                }
+            }
+            var escapeNext = false
+            for (char in line) {
+
+                if (escapeNext) {
+                    escapeNext = false
+                    continue
+                }
+
+                when (char) {
+                    '\\' -> escapeNext = true
+
+                    '{' -> blocksIndicators += CURLY_BRACES
+                    '}' -> blocksIndicators.removeAllElementsFromLast(CURLY_BRACES)
+
+                    '[' -> blocksIndicators += SQUARE_BRACKETS
+                    ']' -> blocksIndicators.removeAllElementsFromLast(SQUARE_BRACKETS)
+
+                    '(' -> blocksIndicators += PARENTHESES
+                    ')' -> blocksIndicators.removeAllElementsFromLast(PARENTHESES)
+
+                    '<' -> blocksIndicators += ANGULAR_BRACKETS
+                    '>' -> blocksIndicators.removeAllElementsFromLast(ANGULAR_BRACKETS)
+
+                    '`' -> if (!blocksIndicators.removeAllElementsFromLast(BACKTICKS)) blocksIndicators += BACKTICKS
+
+                    // TODO html tags
```

Comment:
Could you create issues and convert all TODOs to the TODO: issue-1365 for example or remove

I just share with your Apache Foundation principles, you could follow them or not, your choice

Reply:
good idea!